app/src/test/java/org/oppia/android/app/activity/route/ActivityRouterTest.kt

```diff
@@ -0,0 +1,208 @@
+import org.oppia.android.testing.robolectric.RobolectricModule
+import org.oppia.android.testing.threading.TestDispatcherModule
+import org.oppia.android.testing.time.FakeOppiaClockModule
+import org.oppia.android.util.accessibility.AccessibilityTestModule
+import org.oppia.android.util.caching.AssetModule
+import org.oppia.android.util.caching.testing.CachingTestModule
+import org.oppia.android.util.extensions.getProtoExtra
+import org.oppia.android.util.gcsresource.GcsResourceModule
+import org.oppia.android.util.locale.LocaleProdModule
+import org.oppia.android.util.logging.EventLoggingConfigurationModule
+import org.oppia.android.util.logging.LoggerModule
+import org.oppia.android.util.logging.SyncStatusModule
+import org.oppia.android.util.logging.firebase.FirebaseLogUploaderModule
+import org.oppia.android.util.networking.NetworkConnectionDebugUtilModule
+import org.oppia.android.util.networking.NetworkConnectionUtilDebugModule
+import org.oppia.android.util.parser.html.HtmlParserEntityTypeModule
+import org.oppia.android.util.parser.image.GlideImageLoaderModule
+import org.oppia.android.util.parser.image.ImageParsingModule
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Singleton
+
+private const val internalProfileId = 1
+
+/** Tests for [ActivityRouter]. */
+// FunctionName: test names are conventionally named with underscores.
+@Suppress("FunctionName")
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(application = ActivityRouterTest.TestApplication::class)
+class ActivityRouterTest {
+  @get:Rule
+  val initializeDefaultLocaleRule = InitializeDefaultLocaleRule()
+
+  @get:Rule
+  var activityRule =
+    ActivityScenarioRule<TestActivity>(
+      TestActivity.createIntent(ApplicationProvider.getApplicationContext())
+    )
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+  }
+
+  @Test
+  fun testActivityRouter_routeToDestination() {
+    val activityRouter = retrieveActivityRouter()
+    activityRule.scenario.onActivity {
+      val recentlyPlayedActivityParams =
+        RecentlyPlayedActivityParams
+          .newBuilder()
+          .setProfileId(ProfileId.newBuilder().setInternalId(internalProfileId).build())
+          .setActivityTitle(RecentlyPlayedActivityTitle.RECENTLY_PLAYED_STORIES).build()
+      val intent = activityRouter.routeToScreen(
+        DestinationScreen
+          .newBuilder()
+          .setRecentlyPlayedActivityParams(recentlyPlayedActivityParams)
+          .build()
+      )
+      assertThat(intent).hasComponentClass(RecentlyPlayedActivity::class.java)
+      assertThat(
+        intent?.getProtoExtra(
+          RecentlyPlayedActivity.RECENTLY_PLAYED_ACTIVITY_INTENT_EXTRAS_KEY,
+          RecentlyPlayedActivityParams.getDefaultInstance()
+        )
+      ).isEqualTo(recentlyPlayedActivityParams)
+    }
+  }
+
+  private fun retrieveActivityRouter(): ActivityRouter {
```

Comment:
Is this actually needed? The test above is already using onActivity, so the extra helper method doesn't seem needed. Could we just retrieve the utility directly in the onActivity lambda?

Reply:
Removed it thanks